storage/interfaces.go

```diff
@@ -214,20 +216,14 @@ type Visibility interface {
 	// Get retrieves a visibility using the provided id from SM DB
 	Get(ctx context.Context, id string) (*types.Visibility, error)
 
-	// ListByPlatformID retrieves all visibilities from SM DB that match the specified platform id
-	ListByPlatformID(ctx context.Context, platformID string) ([]*types.Visibility, error)
-
-	// ListByServicePlanID retrieves all visibilities from SM DB that match the specified service plan id
-	ListByServicePlanID(ctx context.Context, servicePlanID string) ([]*types.Visibility, error)
-
 	// List retrieves all visibilities from SM DB
-	List(ctx context.Context) ([]*types.Visibility, error)
+	List(ctx context.Context, criteria ...query.Criterion) ([]*types.Visibility, error)
 
 	// Delete deletes a visibility from SM DB
-	Delete(ctx context.Context, id string) error
+	Delete(ctx context.Context, criteria ...query.Criterion) error
 
 	// Update updates a visibility from SM DB
-	Update(ctx context.Context, visibility *types.Visibility) error
+	Update(ctx context.Context, visibility *types.Visibility, labelChanges ...*query.LabelChange) error
```

Comment:
not sure if this is very user friendly - in case we want to introduce label changes only we would need to first go Get to obtain the resource and then use this method

Reply:
That's what the initial idea of the labels was - they are not updatable by themselves, but only via the entity they're labelling. Moreover, you might have the same label (key-value) labelling different visibilities, so the only way to update the label for the first visibility is to either know the label's id or the associated visibility